alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteMonitorAction.kt

```diff
@@ -123,6 +126,34 @@ class TransportDeleteMonitorAction @Inject constructor(
             }
         }
 
+        /**
+         * Checks if the monitor is part of the workflow
+         *
+         * @param monitorId id of monitor that is checked if it is a workflow delegate
+         */
+        private suspend fun monitorIsNotInWorkflows(monitorId: String): Boolean {
+            val queryBuilder = QueryBuilders.nestedQuery(
+                Workflow.WORKFLOW_DELEGATE_PATH,
+                QueryBuilders.boolQuery().must(
+                    QueryBuilders.matchQuery(
+                        Workflow.WORKFLOW_MONITOR_PATH,
+                        monitorId
+                    )
+                ),
+                ScoreMode.None
+            )
+
+            val searchRequest = SearchRequest()
+                .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
+                .source(SearchSourceBuilder().query(queryBuilder).fetchSource(true))
+
+            val searchResponse: SearchResponse = client.suspendUntil { search(searchRequest, it) }
+            if (searchResponse.hits.totalHits?.value == 0L) {
+                return true
+            }
+            return false
```

Comment:
You can see [here](https://github.com/opensearch-project/alerting/pull/835/files#diff-b683c9727a88a6431493993e071ffff2cc7bb80cba50fa4b9e28d4baba10d5c9R109-R112) - if the canDelete is false, user is getting the message that the monitor can't be deleted. The question is should we be more descriptive or it's enough to say that user is not allowed to do deleting. 
Now I just saw Ashish comment - where he said that we should tell to end user that the monitor can't be deleted because it is a part of the workflow